src/routes/index.ts

```diff
@@ -0,0 +1,45 @@
+import express, { NextFunction, Request, Response, Router } from 'express';
+import ContactRoutes from './contact-requests';
+import { HTTPError } from '../utils/http-error';
+
+// For this API, it is necessary to enable multiple routes to get or send different data.
+const router: Router = express.Router();
+export default router;
+
+// If a GET request is made with no route specified, return 'Code:You API Demo'.
+router.get('/', (req: Request, res: Response) => {
+    res.send('Code:You API Demo');
+});
+
+router.use('/contact-requests', ContactRoutes);
+
+/*
+    This bit will catch any errors thrown in our routes.
+    We can then inspect them to see if they are of type HTTPError, and respect the proposed code if
+    it is!
+
+    Either way - we can send the stack trace if it exists on the error, and use message as a backup.
+
+    If all else fails - we send a generic error response.
+ */
+router.use((error: unknown, req: Request, res: Response, next: NextFunction) => {
```

Comment:
Makes use of `express-async-error` dep to catch and respond with error data.

In real apps - you may want to avoid passing the stacktrace, but for instructional purposes this is ideal.

We can make use of `HTTPError` to allow the application to specify the error code at any level we want, and this allows us to avoid `IResponse` and we can more cleanly write the "intended" code elsewhere

Reply:
Yeah, I wish I knew about that middle ware from the beginning. I should have had a little more R and a little less D when I was doing the R&D here haha. Great call out.

It may be good to mention exactly what you say here in the comment as well. "In real apps - you may want to avoid passing the stacktrace, but for instructional purposes this is ideal." If you don't feel like making another push, I'll add it after the merge. 😄

Reply:
Added!